doc/config.rst

```diff
@@ -295,6 +295,14 @@ General configuration
 
    .. versionadded:: 1.3
 
+.. confval:: manpages_url_pattern
+
+   A URL to cross-reference manpages directives. If this is defined to
+   ``https://manpages.debian.org/%s``, the ``:manpage:ls(1)`` role
```

Comment:
For example, manpages.org uses only the name part (without section) like http://manpages.org/ls .
So I propose you to use named format arguments like `{full}`, `{name}` and `{section}`.

Reply:
In addition, it would be very useful to add a hyperlink to here onto an introduction of `:manpage:` role.

Reply:
Yeah, I thought about doing a tuple - but how do you parse the manpage? Sometimes people don't put a section, sometimes it's page.section or page(section), it's not really well normalized.

But if you think that's important, i'll try to work on a regex and switch to `format`-style patterns.

Reply:
where should manpage link to anyways?

Reply:
okay, so I got this now:

```python
    def visit_manpage(self, node):
        # type: (nodes.Node) -> None
        self.visit_literal_emphasis(node)
        if self.manpages_url_pattern:
            manpage = ' '.join([str(x) for x in node.children
                                if isinstance(x, nodes.Text)])
            pattern = r'^(?P<full>(?P<name>.+)[\(\.](?P<section>[1-9]\w*)?\)?)$'
            r = re.match(pattern, manpage)
            if r:
                info = r.groupdict()
            else:
                info = {'full': manpage}
            node['refuri'] = self.manpages_url_pattern.format(**info)
            self.visit_reference(node)
```

but my concern is that I have to copy this over the `html5.py` builder again, and then this will be duplicated all over the place. what would be a better place to keep that functionality?

Reply:
Good work. I like it :-)

>what would be a better place to keep that functionality?

I think this is a kind of transform module.
```
# in sphinx/transform/__init__.py
class ManpageTransform(SphinxTransform):
    def apply(self):
        for node in self.document.traverse(addnodes.manpage):
            manpage = ' '.join([str(x) for x in node.children
                                if isinstance(x, nodes.Text)])
            pattern = r'^(?P<full>(?P<name>.+)[\(\.](?P<section>[1-9]\w*)?\)?)$'
            r = re.match(pattern, manpage)
            if r:
                info = r.groupdict()
            else:
                info = {'full': manpage}
            for key, value in info.items():
                node[key] = value

# => Add this transform class to the transforms list in sphinx/io.py
```

This allows you to access `name`, ` section` and `full` as attributes of nodes.
And it also make writers simple.

Reply:
i see, done.